scripts/notifications.coffee

```diff
@@ -0,0 +1,78 @@
+# Description:
+#   Notifications for botgt
+# Commands:
+#   hubot notify <message> - notify everyone with something
+#   hubot notify #channel1 ... #channeln !medium1 ... !mediumn <message>  - notify certain channels with something channels and mediums are optional but order matters (do not mix channels and mediums)
+# Author:
+#   Jacob Zipper
+
+request = require "request-promise"
+adminKey = (Buffer.from process.env.BUZZER_ADMIN_KEY_SECRET).toString 'base64'
+options = {
+  uri: 'https://buzzer.hack.gt/graphql',
+  qs: {
+    query: '',
+    variables: ''
+  },
+  headers: {
+    'Authorization': 'Basic ' + adminKey
+  },
+  json: true
+}
+
+module.exports = (robot) ->
+  robot.respond /notify (.*)/i, (res) ->
+    tokens = res.match[1].split " "
+    mediums = []
+    vars = {
+      msg: [],
+      plugins: {}
+    }
+    varsTemp = {
+      twitter: {},
+      facebook: {},
+      slack: {
+        channels: [],
+      },
+      live_site: {}
+    }
+    query = """
+    query ($msg: String!, $plugins: PluginMaster!) {
+      send_message(message: $msg, plugins: $plugins) {
+        plugin
+        errors {
+          error
+          key
+          message
+        }
+      }
+    }
+    """
+    foundChannels = false
+    foundMediums = false
+    for i in [0 ... tokens.length]
+      if !foundChannels
+        if tokens[i][0] == "#"
+          varsTemp.slack.channels.push tokens[i].substr 1
+        else
+          foundChannels = true
+      if foundChannels
+        if !foundMediums
+          if tokens[i][0] == "!"
+            mediums.push tokens[i].substr 1
+          else
+            foundMediums = true
+      if foundChannels && foundMediums
+        vars.msg.push tokens[i]
+      i++
+    vars.msg = vars.msg.join " "
+    for medium in mediums
+      vars.plugins[medium] = varsTemp[medium]
+    options.qs.query = query
+    options.qs.variables = JSON.stringify vars
+    console.log JSON.stringify vars
```

Comment:
Also maybe not an MVP thing, but this is a great place for https://github.com/hackgt/metrics integration. Simply print to STDOUT a single line JSON formatted message in the format:
```
{
    hackgtmetricsversion: 1,
    serviceName: string
    values: object
    tags: object
}
```
 Tags are commonly queried things the database should index (such as medium/channel/person that sent the message) and values are things unique to the message that we should rarely have to query by (such as message content). One example:
```
{
    hackgtmetricsversion: 1,
    serviceName: "botgt-buzzer",
    values: {
        message: "hello world"
    },
    tags: {
        mediums: ['slack', 'twilio']
        sender: ['@ehsanmasdar']
    }
}
```
After this is done, we can visualize and query notifications data in Grafana!

Reply:
Really cool. Will do.